File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        '''
        Given an array of integers and a target
        Returns the indices of two numbers in the array that add up to the target
        '''
        required = {} #Check for remaining value
        for i in range(len(nums)): #Check n-1 values
            if target - nums[i] in required: #If needed value is in array, return
                return [required[target - nums[i]], i]
            else: #Else add to checked array for later comparison at spot
                required[nums[i]] = i

File: two-sum/two-sum.java
import java.util.HashMap;

class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        //Create hashmap for combinations
        HashMap<Integer, Integer> possible = new HashMap<Integer, Integer>();
        
        //Create array to store answer
        int[] answer = {-1, -1};
        
        for (int i = 0; i < nums.length; i++)
        {
            if (possible.containsKey(target - nums[i])) //Check if complement found
            {
                answer[0] = possible.get(target-nums[i]);
                answer[1] = i;
                break;
            }
            else // Add to hashmap for later
            {
                possible.put(nums[i], i);
            }
        }
        return answer;
    }
}